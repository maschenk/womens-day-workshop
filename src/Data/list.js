const list = {
  title: 'Make a to do list work!',
  items: [
    { itemText: 'Show the items in the list', checked: false },
    { itemText: 'Check off an item', checked: false },
    { itemText: 'Add a new item to the list', checked: false },
    { itemText: 'Optional: make it look like I want to', checked: false }
  ]
};

export default list;
