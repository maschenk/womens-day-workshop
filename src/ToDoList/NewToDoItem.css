.add-button {
  display: inline-flex;
  box-sizing: border-box;
  height: 2rem;
  margin-left: 20px;
  min-width: 7.5rem;
  justify-content: center;
  align-items: center;
  padding: 0 1rem;
  border: none;
  border-radius: .25rem;
  background-color: #ff790f;
  color: #fff;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  letter-spacing: .025rem;
  transition: box-shadow .1s,background-color .1s;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
}