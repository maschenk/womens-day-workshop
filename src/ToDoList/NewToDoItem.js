import React from 'react';

class NewToDoItem extends React.Component {
  constructor() {
    super();
    this.state = { text: '' };

    this.handleAdd = this.handleAdd.bind(this);
    this.updateText = this.updateText.bind(this);
  }

  handleAdd(event) {
    event.preventDefault();
    const { text: newItemText } = this.state;
    const { onAdd } = this.props;

    onAdd(newItemText);

    this.setState({ text: '' });
  }

  updateText(event) {
    event.preventDefault();
    this.setState({ text: event.target.value });
  }

  render() {
    return (
      <div data-testid="new-to-do-item">
        <input
          type="text"
          onInput={this.updateText}
          value={this.state.text}
          data-testid="item-input"
        />

        <button type="button" onClick={this.handleAdd} data-testid="add-button">
          Add To Do
        </button>
      </div>
    );
  }
}

export default NewToDoItem;
