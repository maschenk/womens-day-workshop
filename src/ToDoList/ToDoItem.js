import React from 'react';

const ToDoItem = ({ item: { itemText, checked }, onCheck, index }) => {
  const handleCheck = () => {
    onCheck(index);
  };

  return (
    <li data-testid="to-do-item">
      <input
        type="checkbox"
        checked={checked}
        onChange={handleCheck}
        data-testid="item-checkbox"
      />{' '}
      {itemText}
    </li>
  );
};

export default ToDoItem;
