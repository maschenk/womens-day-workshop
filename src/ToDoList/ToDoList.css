.list-container {
  /* positioning */
  display: flex;
  flex-direction: column;
  align-items: center;

  /* sizing */
  min-height: 40vh;
  width: 50%;

  /* styling */
  background-color: lightcoral;
}

.list-title {
  font-size: large;
}
