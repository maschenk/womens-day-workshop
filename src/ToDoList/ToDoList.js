import React from 'react';
import './ToDoList.css';
import ToDoItem from './ToDoItem';
import NewToDoItem from './NewToDoItem';

class ToDoList extends React.Component {
  constructor() {
    super();
    this.state = { items: [] };

    this.addItem = this.addItem.bind(this);
    this.checkItem = this.checkItem.bind(this);
  }

  componentDidMount() {
    const {
      list: { items }
    } = this.props;

    this.setState({ items: items });
  }

  addItem(newItemText) {
    const { items } = this.state;
    items.push({
      itemText: newItemText,
      checked: false
    });
    console.log(items);

    this.setState({ items: items });
  }

  checkItem(index) {
    const { items } = this.state;

    const oldStatus = items[index].checked;

    items[index].checked = !oldStatus;

    this.setState({ items: items });
  }

  render() {
    const { items } = this.state;
    const {
      list: { title }
    } = this.props;

    return (
      <div className="list-container" data-testid="list-container">
        <h4 className="list-title" data-testid="list-title">
          {title}
        </h4>
        <ul className="list-list" data-testid="list-list">
          {items &&
            items.map((item, index) => (
              <ToDoItem
                key={item.itemText}
                item={item}
                onCheck={this.checkItem}
                index={index}
              />
            ))}
        </ul>
        <NewToDoItem onAdd={this.addItem} />
      </div>
    );
  }
}

export default ToDoList;
