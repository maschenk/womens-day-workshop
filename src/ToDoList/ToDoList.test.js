import { render } from '@testing-library/react';
import ToDoList from './ToDoList';
import list from '../Data/list';

test('should render', () => {
  const { container } = render(<ToDoList list={list} />);
  expect(container).toMatchSnapshot();
});
