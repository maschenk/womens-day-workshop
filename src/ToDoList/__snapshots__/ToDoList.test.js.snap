// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="list-container"
    data-testid="list-container"
  >
    <h4
      class="list-title"
      data-testid="list-title"
    >
      Make a to do list work!
    </h4>
    <ul
      class="list-list"
      data-testid="list-list"
    >
      <li
        data-testid="to-do-item"
      >
        <input
          data-testid="item-checkbox"
          type="checkbox"
        />
         
        Show the items in the list
      </li>
      <li
        data-testid="to-do-item"
      >
        <input
          data-testid="item-checkbox"
          type="checkbox"
        />
         
        Check off an item
      </li>
      <li
        data-testid="to-do-item"
      >
        <input
          data-testid="item-checkbox"
          type="checkbox"
        />
         
        Add a new item to the list
      </li>
      <li
        data-testid="to-do-item"
      >
        <input
          data-testid="item-checkbox"
          type="checkbox"
        />
         
        Optional: make it look like I want to
      </li>
    </ul>
    <div
      data-testid="new-to-do-item"
    >
      <input
        data-testid="item-input"
        type="text"
        value=""
      />
      <button
        data-testid="add-button"
        type="button"
      >
        Add a new Item
      </button>
    </div>
  </div>
</div>
`;
